From 6374231228a099be8cfa8caeeaa94624550a7253 Mon Sep 17 00:00:00 2001
From: "shanwu.hu" <shanwu.hu@amlogic.com>
Date: Tue, 12 Aug 2014 16:46:53 +0800
Subject: [PATCH 4759/5965]  PD #93541: fix compiling error/warning of display

Conflicts:

	drivers/amlogic/display/backlight/aml_bl_extern/pmu_aml1218.c

Change-Id: I71e8acc103e19961b48b54aa8ce929fdb9a28f0d
---
 .../display/aml_logo/logo_display/output_dev_osd.c        | 8 ++++++++
 drivers/amlogic/display/aml_logo/logo_parse/jpeg_parser.c | 4 ++++
 drivers/amlogic/display/backlight/aml_bl.c                | 3 ++-
 .../amlogic/display/backlight/aml_bl_extern/pmu_aml1218.c | 2 +-
 drivers/amlogic/display/osd/osd_hw.c                      | 4 ++++
 drivers/amlogic/display/osd/osd_main.c                    | 2 ++
 drivers/amlogic/display/vout/enc_clk_config.c             | 2 ++
 include/linux/amlogic/aml_bl_extern.h                     | 2 +-
 8 files changed, 24 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/display/aml_logo/logo_display/output_dev_osd.c b/drivers/amlogic/display/aml_logo/logo_display/output_dev_osd.c
index c3274ed40182..c372bec24402 100755
--- a/drivers/amlogic/display/aml_logo/logo_display/output_dev_osd.c
+++ b/drivers/amlogic/display/aml_logo/logo_display/output_dev_osd.c
@@ -208,8 +208,12 @@ void set_osd_freescaler(int index, logo_object_t *plogo, vmode_t new_mode) {
 
 static int osd0_init(logo_object_t *plogo)
 {
+#if defined(CONFIG_AM_HDMI_ONLY)
 	int hpd_state = 0;
+#endif
+#if defined(CONFIG_AM_HDMI_ONLY) || (MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8)
 	vmode_t cur_mode = plogo->para.vout_mode;
+#endif
 
 	if(plogo->para.output_dev_type==output_osd0.idx)
 	{
@@ -267,8 +271,12 @@ static int osd0_init(logo_object_t *plogo)
 }
 static int osd1_init(logo_object_t *plogo)
 {
+#if defined(CONFIG_AM_HDMI_ONLY)
 	int hpd_state = 0;
+#endif
+#if defined(CONFIG_AM_HDMI_ONLY) || (MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8)
 	vmode_t cur_mode = plogo->para.vout_mode;
+#endif
 
 	if(plogo->para.output_dev_type==output_osd1.idx)
 	{
diff --git a/drivers/amlogic/display/aml_logo/logo_parse/jpeg_parser.c b/drivers/amlogic/display/aml_logo/logo_parse/jpeg_parser.c
index 1e7fbaf7b2cd..04ec7838d97e 100755
--- a/drivers/amlogic/display/aml_logo/logo_parse/jpeg_parser.c
+++ b/drivers/amlogic/display/aml_logo/logo_parse/jpeg_parser.c
@@ -55,7 +55,9 @@ static const struct vframe_operations_s jpeglogo_vf_provider_op =
     	.get  = jpeglogo_vf_get,
     	.put  = NULL,
 };
+#ifdef CONFIG_AM_VIDEO 
 static struct vframe_provider_s jpeglogo_vf_prov;
+#endif
 static inline u32 index2canvas(u32 index)
 {
     const u32 canvas_tab[4] = {
@@ -342,7 +344,9 @@ static inline void feed_vb(s32 s)
 
 static int hardware_init(logo_object_t *plogo,int logo_size)
 {
+#ifdef CONFIG_AM_STREAMING
 	u32	*mc_addr_aligned = (u32 *)vmjpeg_mc;
+#endif
 	int ret = 0;
 	if(plogo->para.output_dev_type  <=LOGO_DEV_VID ) //now only support display on video layer.
 	{
diff --git a/drivers/amlogic/display/backlight/aml_bl.c b/drivers/amlogic/display/backlight/aml_bl.c
index 723d247effe3..9e0a7574dd00 100755
--- a/drivers/amlogic/display/backlight/aml_bl.c
+++ b/drivers/amlogic/display/backlight/aml_bl.c
@@ -470,7 +470,7 @@ void bl_power_off(int bl_flag)
     printk("backlight power off\n");
     mutex_unlock(&bl_power_mutex);
 }
-
+#ifndef CONFIG_USE_OF
 static DEFINE_MUTEX(bl_level_mutex);
 static void set_backlight_level(unsigned level)
 {
@@ -672,6 +672,7 @@ unsigned get_backlight_level(void)
     return bl_level;
 }
 #endif
+#endif
 
 struct aml_bl {
     const struct aml_bl_platform_data   *pdata;
diff --git a/drivers/amlogic/display/backlight/aml_bl_extern/pmu_aml1218.c b/drivers/amlogic/display/backlight/aml_bl_extern/pmu_aml1218.c
index 9004df253f16..ebb6af74338f 100755
--- a/drivers/amlogic/display/backlight/aml_bl_extern/pmu_aml1218.c
+++ b/drivers/amlogic/display/backlight/aml_bl_extern/pmu_aml1218.c
@@ -167,9 +167,9 @@ static ssize_t bl_extern_debug(struct class *class, struct class_attribute *attr
 {
 #ifdef CONFIG_AMLOGIC_BOARD_HAS_PMU
     struct aml_pmu_driver *pmu_driver;
-#endif
     unsigned char temp;
     unsigned int t[2];
+#endif
     int ret = 0;
 
 #ifdef CONFIG_AMLOGIC_BOARD_HAS_PMU
diff --git a/drivers/amlogic/display/osd/osd_hw.c b/drivers/amlogic/display/osd/osd_hw.c
index b6af907380ed..1551c981bb86 100755
--- a/drivers/amlogic/display/osd/osd_hw.c
+++ b/drivers/amlogic/display/osd/osd_hw.c
@@ -217,7 +217,9 @@ static const struct vframe_operations_s osd_vf_provider =
 	.put  = NULL,
 };
 
+#ifdef CONFIG_AM_VIDEO
 static unsigned char osd_vf_prov_init = 0;
+#endif
 
 static inline void  osd_update_3d_mode(int enable_osd1,int enable_osd2)
 {
@@ -237,6 +239,7 @@ static inline void wait_vsync_wakeup(void)
 	wake_up_interruptible(&osd_vsync_wq);
 }
 
+#ifdef CONFIG_VSYNC_RDMA
 static irqreturn_t osd_rdma_isr(int irq, void *dev_id)
 {
 #define  	VOUT_ENCI	1
@@ -341,6 +344,7 @@ static irqreturn_t osd_rdma_isr(int irq, void *dev_id)
 
 	return IRQ_HANDLED;
 }
+#endif
 
 static inline void  walk_through_update_list(void)
 {
diff --git a/drivers/amlogic/display/osd/osd_main.c b/drivers/amlogic/display/osd/osd_main.c
index 63ed94980111..f24ddb156f61 100755
--- a/drivers/amlogic/display/osd/osd_main.c
+++ b/drivers/amlogic/display/osd/osd_main.c
@@ -1589,8 +1589,10 @@ osd_probe(struct platform_device *pdev)
 	int  logo_osd_index=0,i;
 	myfb_dev_t 	*fbdev = NULL;
 	vmode_t current_mode = VMODE_MASK;
+#ifdef CONFIG_AM_HDMI_ONLY
 	vmode_t cvbs_mode = VMODE_MASK;
 	int hpd_state = 0;
+#endif
 	const void *prop;
 	int prop_idx=0;
 	int rotation = 0;
diff --git a/drivers/amlogic/display/vout/enc_clk_config.c b/drivers/amlogic/display/vout/enc_clk_config.c
index a5d21385ae68..4b03d75ee636 100755
--- a/drivers/amlogic/display/vout/enc_clk_config.c
+++ b/drivers/amlogic/display/vout/enc_clk_config.c
@@ -466,6 +466,7 @@ static void set_vdac1_div(unsigned div)
 
 static enc_clk_val_t setting_enc_clk_val_m8m2[] = {
 
+#if MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8
 		{VMODE_480I,       2160, 8, 1, 1, VIU_ENCI,  5, 4, 2,-1,  2, -1, -1,  2,  -1},
 		{VMODE_480I_RPT,   2160, 4, 1, 1, VIU_ENCI,  5, 4, 2,-1,  4, -1, -1,  2,  -1},
 		{VMODE_480CVBS,    1296, 4, 1, 1, VIU_ENCI,  6, 4, 2,-1,  2, -1, -1,  2,  -1},
@@ -491,6 +492,7 @@ static enc_clk_val_t setting_enc_clk_val_m8m2[] = {
 		{VMODE_VGA,  1066, 3, 1, 1, VIU_ENCP, 10, 1, 2, 1, -1, -1, -1,  1,  1},
 		{VMODE_SVGA, 1058, 2, 1, 1, VIU_ENCP, 10, 1, 2, 1, -1, -1, -1,  1,  1},
 		{VMODE_XGA, 1085, 1, 1, 1, VIU_ENCP, 5, 1, 1, 1, -1, -1, -1,  1,  1},
+#endif
 };
 static enc_clk_val_t setting_enc_clk_val[] = {
 
diff --git a/include/linux/amlogic/aml_bl_extern.h b/include/linux/amlogic/aml_bl_extern.h
index 3dcc7162df67..05683d9d54d4 100755
--- a/include/linux/amlogic/aml_bl_extern.h
+++ b/include/linux/amlogic/aml_bl_extern.h
@@ -14,7 +14,7 @@ typedef enum {
 
 //global API
 struct aml_bl_extern_driver_t {
-	char *name;
+	const char *name;
 	Bl_Extern_Type_t type;
 	int (*power_on) (void);
 	int (*power_off)(void);
-- 
2.19.0

